### PR TITLE
Model tags support

### PR DESCRIPTION
**Pull request by ekoby**

### CODE REVIEW

File: library/model_support.c [hunk 1]
```diff
@@ -243,6 +276,10 @@ int model_to_json(void *obj, type_meta *meta, int indent, char *buf, size_t maxl
 
             *p++ = ']';
         }
+        else {
+            fprintf(stderr, "unsupported mod[%d] for field[%s]", fm->mod, fm->name);
```
Comment (dovholuknf) [inline library/model_support.c:280]: why not ZITI_LOG ?

File: tests/model_tests.cpp [hunk 2]
```diff
@@ -278,4 +278,74 @@ TEST_CASE("model compare with array", "[model]") {
...
+        }
+    })";
+
+    tagged obj;
+    REQUIRE(parse_tagged(&obj, json, strlen(json)) == 0);
+
+    const char *num = (const char *) model_map_get(&obj.tags, "num");
+    CHECK_THAT(num, Equals("42"));
+    CHECK_THAT((const char *) model_map_get(&obj.tags, "ok"), Equals("true"));
+    CHECK_THAT((const char *) model_map_get(&obj.tags, "msg"), Equals("hello\nworld!"));
```
Comment (dovholuknf) [inline tests/model_tests.cpp:306]: isn't this technically invalid json since it has an embedded newline?

File: library/model_support.c [hunk 3]
```diff
@@ -349,6 +414,53 @@ static int parse_array(void **arr, const char *json, jsmntok_t *tok, type_meta *
...
+        ZITI_LOG(ERROR, "unexspected JSON token near '%.*s', expecting object", 20, json + tok->start);
+        return -1;
+    }
+    model_map *map = mapp;
+    int tokens_processed = 1;
+    int children = tok->size;
+    tok++;
+    for (int i = 0; i < children; i++) {
+        if (tok->type != JSMN_STRING) {
+            ZITI_LOG(ERROR, "parsing[map] error: unexpected token starting at `%.*s'\n", 20, json + tok->start);
```
Comment (dovholuknf) [inline library/model_support.c:428]: wouldn't this \n split the line?

File: library/model_support.c [hunk 4]
```diff
@@ -349,6 +414,53 @@ static int parse_array(void **arr, const char *json, jsmntok_t *tok, type_meta *
     return processed;
 }
 
+static int parse_map(void *mapp, const char *json, jsmntok_t *tok, type_meta *el_meta) {
+    if (tok->type != JSMN_OBJECT) {
+        ZITI_LOG(ERROR, "unexspected JSON token near '%.*s', expecting object", 20, json + tok->start);
```
Comment (dovholuknf) [inline library/model_support.c:419]: i didn't expect to see `unexspected` here

Comment (ekoby) [reply library/model_support.c:419]: it means not up to spec, duh!

**Pull request merged**